cpp/src/arrow/builder-benchmark.cc

```diff
@@ -176,6 +177,26 @@ static void BuildFixedSizeBinaryArray(
   state.SetBytesProcessed(state.iterations() * kBytesProcessed);
 }
 
+static void BuildDecimalArray(benchmark::State& state) {  // NOLINT non-const reference
+  auto type = decimal(10, 5);
+  Decimal128 value;
+  int32_t precision = 0;
+  int32_t scale = 0;
+  ABORT_NOT_OK(Decimal128::FromString("1234.1234", &value, &precision, &scale));
+  for (auto _ : state) {
+    Decimal128Builder builder(type);
+
+    for (int64_t i = 0; i < kRounds * kNumberOfElements; i++) {
+      ABORT_NOT_OK(builder.Append(value));
+    }
+
+    std::shared_ptr<Array> out;
+    ABORT_NOT_OK(builder.Finish(&out));
+  }
+
+  state.SetBytesProcessed(state.iterations() * kBytesProcessed * 2);
```

Comment:
Why `* 2`?

Reply:
Ok, I see... perhaps write this `state.iterations() * kRounds * kNumberOfElements * 16`? Otherwise it's a bit obscure IMHO.

Reply:
Done (I was mostly following the existing pattern).